splink/linker.py

```diff
@@ -1189,6 +1190,12 @@ def deterministic_link(self) -> SplinkDataFrame:
                 represents a table materialised in the database. Methods on the
                 SplinkDataFrame allow you to access the underlying data.
         """
+
+        # Allows clustering during a deterministic linkage.
+        # This is used in `cluster_pairwise_predictions_at_threshold`
+        # to set the cluster threshold to 1
+        self._deterministic_link_mode = True
```

Comment:
Do we definitely want this to remain as `True` if this is run?

I would imagine we do, but it doesn't hurt to ask.

Reply:
What are the other options other than it remaining as true? As in finding a way to revert to `False` once it has run this link?

Reply:
This would break if someone decided to reload their settings object -> `linker.predict()`
or if they submitted a valid settings object (BRs + CLs) and decided to:
`linker.deterministic_link()` -> `linker.predict()`

Reply:
I don't see anyone doing this in reality, but I might as well flag it

Reply:
I guess the only issue here could be if a user ran a deterministic link off their linker and then used the same linker to try and do a probabilistic link without instantiating again. I reckon that is a pretty odd edge case that I don't think we should worry about too much